skfda/tests/test_pace.py

```diff
@@ -0,0 +1,607 @@
+    return tuple(request.param)
+
+##############################################################################
+# Tests
+##############################################################################
+
+
+def test_raises(pace_raises_fixture: tuple[dict[str, Any], str]) -> None:
+    """Check that PACE init raises ValueError."""
+    kwargs, expected_msg = pace_raises_fixture
+    with pytest.raises(ValueError, match=expected_msg):
+        PACE(**kwargs)
+
+def test_input_fd_adequate(
+    input_fd_raises_fixture: tuple[PACE, FDataIrregular, type[Exception], str],
+) -> None:
+    """Check that the number of components is valid."""
+    pace, fd, excep, error_msg = input_fd_raises_fixture
+    with pytest.raises(excep, match=error_msg):
+        pace.fit(fd)
+
+def test_mean_from_matlab(
+    matlab_mean_data: NDArrayFloat,
+    fetch_cd4_fixture: FDataIrregular,
+) -> None:
+    """
+    Check that the mean is calculated correctly.
+
+    Check that the mean is calculated correctly, comparing it with the result
+    obtained from the matlab PACE package. The precalculated result is obtained
+    by using the following configuration:
+        options = setOptions(...
+            'yname', 'CD4', ...
+            'regular', 0, ...                       % Irregular data
+            'kernel', 'gauss', ...                  % Gaussian kernel
+            'bwmu', 0, ...                          % Estimate the bw
+            'bwmu_gcv', 2, ...                      % Calculate using GCV
+            'bwxcov', [0,0], ...                    % Estimate the bw
+            'bwxcov_gcv', 1, ...                    % Calculate using GCV
+            'selection_k', 'fixed', ...             % Fix number of components
+            'method', 'CE', ...                     % Conditional Expectation
+            'selection_k', 'FVE', ...               % Fraction of Variance Exp.
+            'FVE_threshold',0.99, ...               % Eigenfunctions .99
+            'screePlot', 1, ...                     % Generate plot to show FVE
+            'verbose', 'on', ...                    % Inform of choices
+            'error', 1 ...                          % Assume the data is noisy
+        );
+    """
+    pace = PACE(
+        n_components=0.99,
+        n_grid_points=51,
+        bandwidth_mean=8.25,
+        bandwidth_cov=6.269,
+    )
+    pace.fit(fetch_cd4_fixture)
+
+    np.testing.assert_allclose(
+        matlab_mean_data,
+        pace.mean_.data_matrix[0, :, 0],
+        atol=1e-7,
+    )
+
+def test_cov_from_matlab(
+    matlab_cov_data: NDArrayFloat,
+    fetch_cd4_fixture: FDataIrregular,
+) -> None:
+    """
+    Check that the covariance is calculated correctly.
+
+    Check that the covariance is calculated correctly, comparing it with the
+    resul obtained from the matlab PACE package. The precalculated result is
+    obtained by using the following configuration:
+        options = setOptions(...
```

Comment:
I would add also the code you used. The future programmers (me, probably) would thank you for that.